File: app/src/main/java/com/example/practice/adapter/MainArticleAdapter.java
package com.example.practice.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.practice.R;
import com.example.practice.bean.MainArticleBean;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

/**
 * Created By 大苏打
 * on 2020/9/21
 */
public class MainArticleAdapter extends RecyclerView.Adapter<MainArticleAdapter.MyViewHolder>{

    List<MainArticleBean> list;
    Context context;

    public MainArticleAdapter(List<MainArticleBean> list, Context context){
        this.list = list;
        this.context = context;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType){
        View inflate = LayoutInflater.from(context).inflate(R.layout.item_main_article, parent, false);
        return new MyViewHolder(inflate);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position){
        holder.tv_title.setText(list.get(position).getTitle());
        holder.tv_chaptername.setText("分类："+list.get(position).getSuperChapterName()+"/"+list.get(position).getChapterName());
        holder.tv_shareuser.setText("分享人："+list.get(position).getShareUser());
        holder.tv_time.setText("时间："+list.get(position).getNiceDate());
    }

    @Override
    public int getItemCount(){
        return list.size();
    }

    public class MyViewHolder extends RecyclerView.ViewHolder{
        TextView tv_title;
        TextView tv_shareuser;
        TextView tv_chaptername;
        TextView tv_time;
        public MyViewHolder(@NonNull View itemView){
            super(itemView);
            tv_title=itemView.findViewById(R.id.tv_title);
            tv_shareuser=itemView.findViewById(R.id.tv_shareuser);
            tv_chaptername=itemView.findViewById(R.id.tv_chaptername);
            tv_time=itemView.findViewById(R.id.tv_time);
        }
    }
}

File: app/src/main/java/com/example/practice/adapter/MainArticleAdapter.kt
package com.example.practice.adapter

import android.text.TextUtils
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.viewholder.BaseViewHolder
import com.example.practice.R
import com.example.practice.bean.MainArticleBean


/**
 * Created By 大苏打
 * on 2020/9/21
 */
class MainArticleAdapter (layoutResId:Int, data:MutableList<MainArticleBean>) :
    BaseQuickAdapter<MainArticleBean, BaseViewHolder>(layoutResId, data) {
    override fun convert(holder: BaseViewHolder, mainArticleBean: MainArticleBean) {
        holder.setText(R.id.tv_title, mainArticleBean.title)
        holder.setText(R.id.tv_chaptername, "分类：" + mainArticleBean.chapterName)
        holder.setText(
            R.id.tv_shareuser,
            if (TextUtils.isEmpty(mainArticleBean.shareUser)) "作者：" + mainArticleBean.author else "分享人：" + mainArticleBean.shareUser
        )
        holder.setText(R.id.tv_time, "时间：" + mainArticleBean.niceDate)
    }
}
